### PR TITLE
3x backport - Port various elixir tests

### PR DESCRIPTION
**Pull request by jjrodrig**
## Overview
This PR backports into 3.x the following test ports

 - rewrite and rewrite_js (#3174)
 - show_documents and list_views (#3179)
 - security_validation (#3180)

A small fix identified during rewrite_js porting is also included (#3172)

## Testing recommendations
```
 make elixir
```

## Related Issues or Pull Requests
See overview

## Checklist

- [X] Code is written and works correctly
- [X] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/chttpd/src/chttpd_rewrite.erl [hunk 1]
```diff
@@ -71,8 +71,9 @@ do_rewrite(#httpd{mochi_req=MochiReq}=Req, {Props}=Rewrite) when is_list(Props)
                 undefined -> erlang:get(mochiweb_request_body);
                 B -> B
             end,
+            NewMochiReq:cleanup(),
             case Body of
-                undefined -> NewMochiReq:cleanup();
+                undefined -> [];
                 _ -> erlang:put(mochiweb_request_body, Body)
```
Comment (dottorblaster) [inline src/chttpd/src/chttpd_rewrite.erl:77]: Are you sure this has to be backported? It's not elixir

Comment (jjrodrig) [reply src/chttpd/src/chttpd_rewrite.erl:77]: Yes, it is a minor fix detected during the rewrite_js porting. rewrite_js_test.exs will fail without this fix.

It is already merged into master

Comment (dottorblaster) [reply src/chttpd/src/chttpd_rewrite.erl:77]: Ok then!

**Review by dottorblaster — APPROVED** [review]

**Comment by jjrodrig** [comment]
Thanks!

**Pull request merged**